client/src/pages/Register/PaymentStep.tsx

```diff
@@ -1,60 +1,85 @@
 import React from 'react';
+import { useHistory } from 'react-router-dom';
 import type {
   PayPalButtonsComponentOptions,
   SHIPPING_PREFERENCE,
   FUNDING_SOURCE,
+  PayPalScriptOptions,
+  OrderResponseBody,
 } from '@paypal/paypal-js';
+
+import {
+  PayPalScriptProvider,
+} from '@paypal/react-paypal-js';
+
 import debug from 'utils/debug';
-import { Spinner } from 'react-bootstrap';
 import jsonLogic from 'json-logic-js';
+import Spinner from 'components/Spinner';
 import JsonSchemaForm from 'components/JsonSchemaForm';
+import api, {
+  PaymentType,
+  DepositType,
+} from 'store/register/api';
+import {
+  useRegFormData,
+  setFinalStep,
+  setPaymentInfo,
+  useAppDispatch,
+} from 'store/register/store';
 
-import type { PaymentType } from './index';
 import checkImage from './check-image.png';
 import PayPalButtons from './PayPalButtons';
-import type { RegisterStepProps } from './component';
+import PageWrapper from './PageWrapper';
+
+import './styles.scss';
 
 export type PayPalCreateOrder = NonNullable<PayPalButtonsComponentOptions['createOrder']>;
 export type PayPalOnApprove = (a: FUNDING_SOURCE) => PayPalButtonsComponentOptions['onApprove'];
 
-function PaymentStep(props: RegisterStepProps) {
+function PaymentStep() {
+  const registrationApi = api.useGetRegistrationQuery();
+  const [createInitialPayment] = api.useCreateInitialPaymentMutation();
+  const regFormData = useRegFormData();
+  const dispatch = useAppDispatch();
   const [loading, setLoading] = React.useState(false);
-  const [depositChoice, setDepositChoice] = React.useState<{ deposit: any } | undefined>();
-  if (props.step !== 'payment') return null;
-
-  const paymentData = {
-    type: 'payment',
-    total: (props.totals.total || 0).toFixed(2),
-  };
-
-  if (depositChoice?.deposit) {
-    try {
-      debug('depositChoice', props.totals);
-
-      const values = JSON.parse(depositChoice.deposit);
-
-      const results = jsonLogic.apply(values.logic, props.totals);
-
-      paymentData.type = values.name;
-      paymentData.total = results;
-    } catch (e) {
-      console.error('deposit logic error', e);
-    }
-  }
-
+  const [depositChoice, setDepositChoice] = React.useState<{ deposit: any }>();
+  const [total, setTotal] = React.useState<number>(
+    regFormData.paymentStep?.serverPricingResults.total || 0
+  );
+  const history = useHistory();
+
+  if (
+    registrationApi.isFetching ||
+    registrationApi.isLoading ||
+    !registrationApi.data ||
+    !regFormData.paymentStep
```

Comment:
Although this guard works pretty well, it may be better to redirect to `/form` if the paymentStep data isn't present, what do you think?